shim.c

```diff
@@ -1831,6 +1831,59 @@ debug_hook(void)
 	x = 1;
 }
 
+EFI_STATUS
+check_shim_sbat()
+{
+	unsigned int i;
+	EFI_STATUS efi_status;
+	list_t sbat_var_entries;
+	struct sbat sbat = { 0 };
+	struct sbat_entry *entry = NULL;
+	char *sbat_start = (char *)&_sbat;
+	char *sbat_end = (char *)&_esbat;
+	size_t sbat_size = sbat_end - sbat_start;
+
+	char *sbat_data = AllocatePool(sbat_size + 1);
+	if (!sbat_data) {
+		console_print(L"Failed to allocate SBAT buffer\n");
+		return EFI_OUT_OF_RESOURCES;
+	}
+
+	CopyMem(sbat_data, sbat_start, sbat_size);
+	sbat_data[sbat_size] = '\0';
+
+	efi_status = parse_sbat(sbat_data, sbat_size + 1, &sbat);
```

Comment:
Just an observation - I notice that both callers to parse_sbat() pass in size+1 to include the NULL terminator, but the first thing that parse_sbat() does is subtract 1 from the size.

Reply:
Right, I've no idea why parse_sbat() is substracting that 1 from the size...

Reply:
@chrisccoulson deleted the subtraction since the `parse_sbat()` function should use the size passed by the caller, which  generally would be a null-terminated string.